File: interfaceTest/ssros_ams/src/set_fan_pwm_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "test_interfaces/srv/set_fan_pwm.hpp"

#include <memory>

void set_fan_pwm(const std::shared_ptr<test_interfaces::srv::SetFanPWM::Request> request,
          std::shared_ptr<test_interfaces::srv::SetFanPWM::Response>      response)
{
  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Incoming request\npwm: %d", request->pwm);
  response->success = true;
}

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("set_fan_pwm_server");

  rclcpp::Service<test_interfaces::srv::SetFanPWM>::SharedPtr service =
    node->create_service<test_interfaces::srv::SetFanPWM>("set_fan_pwm", &set_fan_pwm);

  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Ready to set fan pwm.");

  rclcpp::spin(node);
  rclcpp::shutdown();
}

File: interfaceTest/ssros_ams_py/ssros_ams_py/set_fan_pwm_server.py
from test_interfaces.srv import SetFanPWM

import rclpy
from rclpy.node import Node

class SetFanPWMServer(Node):

    def __init__(self):
        super().__init__('set_fan_pwm_server')
        self.srv = self.create_service(SetFanPWM, 'set_fan_pwm', self.set_fan_pwm_callback)

    def set_fan_pwm_callback(self, request, response):
        self.get_logger().info('Incoming request\npwm: %d' % (request.pwm))
        response.success = True
        return response

def main(args=None):
    rclpy.init(args=args)
    set_fan_pwm_server = SetFanPWMServer()
    rclpy.spin(set_fan_pwm_server)
    rclpy.shutdown()

if __name__ == '__main__':
    main()
